token/program-2022/src/extension/memo_transfer/mod.rs

```diff
@@ -33,7 +32,7 @@ impl Extension for MemoTransfer {
 }
 
 /// Determine if a memo is required for transfers into this account
-pub fn memo_required(account_state: &StateWithExtensionsMut<Account>) -> bool {
+pub fn memo_required<BSE: BaseStateWithExtensions<S>, S: BaseState>(account_state: &BSE) -> bool {
```

Comment:
BaseState must be Account specifically, though, right?

Reply:
Now that we have `PodAccount`, it can be either `Account` or `PodAccount`.

I can add a check to make sure that `S::ACCOUNT_TYPE` is `AccountType::Account` though to give at least a runtime check, but it should be redundant considering we're fetching the `MemoTransfer` extension, which is only allowed on `Account`s. What do you think?

Reply:
Yeah, I guess it's fine. I just didn't like the feel of losing compile-time type safety, especially on a public method. I suppose the alternative is to add different Account and Mint traits, implemented by `Account`+`PodAccount` and `Mint`+`PodMint` respectively. That seems like far too much work.